awx/main/credential_plugins/hashivault.py

```diff
@@ -93,16 +120,27 @@ def kv_backend(**kwargs):
         except Exception:
             mount_point, path = secret_path, []
         # https://www.vaultproject.io/api/secret/kv/kv-v2.html#read-secret-version
-        response = sess.get(
-            '/'.join([url, mount_point, 'data'] + path).rstrip('/'),
-            params=params,
-            timeout=30
-        )
+        if cacert:
+            response = sess.get(
+                '/'.join([url, mount_point, 'data'] + path).rstrip('/'),
+                params=params,
+                verify=create_temporary_fifo(cacert.encode()) ,
+                timeout=30
+            )
+        else:
+            response = sess.get(
+                '/'.join([url, mount_point, 'data'] + path).rstrip('/'),
+                params=params,
+                timeout=30
+            )
```

Comment:
I have some concern regarding how we've begun to duplicate some of the request building logic for these, _particularly_ the url building, timeout, and other common params. We've had some tricky bug fixes involving these params and so I think the safe thing to do here is to keep them as de-duplicated as possible.

Let's _build up to_ the request args and keyword args like we do in the other plugins:

```python
request_url = '/'.join([url, mount_point, 'data'] + path).rstrip('/')
request_kwargs = {'params': params, 'timeout': 30 }

if cacert:
    request_kwargs['verify'] = create_temporary_fifo(cacert.encode())

response = sess.get(request_url, **request_kwargs)

```